Source/engine.h

```diff
@@ -27,6 +27,16 @@
 
 namespace devilution {
 
+#if defined(__clang__) || !defined(__GNUC__) || __GNUC__ > 7 || (__GNUC__ == 7 && __GNUC_MINOR__ < 0)

```

Comment:
We should probably use a feature testing macro here:

```suggestion
#if defined(__cpp_lib_clamp)
```

https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros

Reply:
oh wow was hoping something like that existed, but couldn't find anything